modules/apps/depot/depot-test/src/testIntegration/java/com/liferay/depot/web/internal/portlet/action/test/UpdateMembershipsMVCActionCommandTest.java

```diff
@@ -109,33 +120,56 @@ public void testProcessActionWithAddGroupDepotIds() throws Exception {
 		Assert.assertTrue(found);
 	}
 
+	@Test(expected = PrincipalException.MustHavePermission.class)
+	public void testProcessActionWithAddGroupDepotIdsWithoutPermissions()
+		throws Exception {
+
+		Organization organization = OrganizationTestUtil.addOrganization();
+
+		User organizationOwnerUser = UserTestUtil.addOrganizationOwnerUser(
+			organization);
+
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+		String originalName = PrincipalThreadLocal.getName();
+
+		PermissionChecker permissionChecker =
+			PermissionCheckerFactoryUtil.create(organizationOwnerUser);
+
+		PermissionThreadLocal.setPermissionChecker(permissionChecker);
+
+		PrincipalThreadLocal.setName(organizationOwnerUser.getUserId());
```

Comment:
This code looks slightly different than the one in `testProcessActionWithDeleteGroupDepotIdsWithoutPermissions` (mostly a missing newline).

For bonus points, why not create a helper function to hide all these details?
```
_withOrganizationOwnerUser(
    organizationOwnerUser -> {
			ReflectionTestUtil.invoke(
				_mvcActionCommand, "_validateGroups",
				new Class<?>[] {ActionRequest.class},
				new MockActionRequest(
					permissionChecker,
					_companyLocalService.getCompany(
						TestPropsValues.getCompanyId()),
					_groupLocalService.getGroup(TestPropsValues.getGroupId()),
					organizationOwnerUser,
					new long[] {_depotEntry.getGroupId()}, null));
    });
```

Also, shouldn't you delete the organization as well?